pkgs/misc/emulators/wine/default.nix

```diff
@@ -44,7 +44,7 @@
   sdlSupport ? false,
   faudioSupport ? false,
   vkd3dSupport ? false,
-  mingwSupport ? false,
+  mingwSupport ? wineRelease != "stable",
```

Comment:
Forgive my ignorance: why is this nice to have on by default?

Reply:
@SFrijters It's nice to have enabled by default to avoid issues with anticheat software. Arch Linux & Fedora also have this enabled by default:

- Fedora: https://src.fedoraproject.org/rpms/wine/blob/8e216ca407b6c0f78f65f36c5b068c6452701e55/f/wine.spec#_116
- Arch Linux: https://github.com/archlinux/svntogit-community/blob/2435e762eacd989c588200d7cf57d8f4fb2e0cf3/trunk/PKGBUILD#L44

Although `mingwSupport` is already enabled by default in the base build: https://github.com/NixOS/nixpkgs/pull/113146, so it shouldn't need to be configured here.

Reply:
Ah, thanks. And I think indeed you are right, I've seen mingw things pop up in my recent builds.
In that case I don't think anything actually changes compared to my earlier attempt, so LGTM then.

Reply:
Could we clarify why only to enable this on staging/unstable but not stable?

Reply:
It is quite unintuitive API to have `wine.override { wineRelease = "staging"; }` differ from `wine` in more than one way.

Reply:
Is there any reason why this is still disabled on stable Wine today?

Reply:
Likely, nobody cared to enable.

Reply:
Note that it's now always enabled for `wineWow64` because the build requires it for that configuration from 9.x onwards.

https://github.com/NixOS/nixpkgs/pull/281905#issuecomment-1899995964
https://github.com/NixOS/nixpkgs/pull/281905/commits/38f7be8b49aef5a16cf62a3a2d12ae47d45b839a

Reply:
Would it be better to always enable it everywhere?
(`mingwSupport ? true`)

Reply:
The `minimal` wine package uses the defaults, so I think it makes sense to keep as many things as possible disabled to make it truly minimal - `base` and `full` enable it. In fact, I would probably also prefer it if `waylandSupport ? false` and instead we set it to `stdenv.isLinux` for `base` and above.